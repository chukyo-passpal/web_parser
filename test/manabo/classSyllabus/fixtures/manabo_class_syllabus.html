<script>
    function requestReferenceUrl() {
        $('.request_reference_url').each(function (idx, value) {
            var target = $(this);
            var title = $(this).html();
            glexa.ajax({
                action: 'addon_syllabus_api_booknumber',
                withoutLoading: true,
                params: {
                    code: $(this).attr('data-code'),
                },
                onSuccess: function (json) {
                    if (json.html) {
                        var url = json.html;
                        target.html('<a href="' + url + '" target="_blank">' + title + '</a>');
                    }
                },
            });
        });
    }
</script>
<div class="panel panel-default" style="height: 40em; overflow: auto;">
    <div class="panel-heading">
        <i class="mark"></i>
        シラバス詳細
    </div>
    <div class="panel-body">
        <div class="table-responsive">
            <table class="table table-default">
                <tr>
                    <th style="width:11em;">授業概要・目的</th>
                    <td colspan="7">
                        「アルゴリズムとデータ構造１」の学習内容を深め、様々なデータ構造と、それらを取り扱う基本的なアルゴリズムの理解を通して、アルゴリズムの設計や解析をするための基本的な手法の修得を指導する。この授業の目標は、実際にコンピュータプログラムを作成する際に、問題に適切なデータ構造とアルゴリズムを利用、選択できるようになることである。授業は講義形式およびプログラミング演習を通して行う。アルゴリズムとデータ構造2では、主にリスト、木などの基本的なデータ構造の利用方法・実現方法を解説する。<br />
                    </td>
                </tr>
                <tr>
                    <th>学修到達目標</th>
                    <td colspan="7">
                        <ul>
                            <li>基本的なデータ構造とその操作アルゴリズムについて、コンピュータモデルと情報数学に基づく説明と分析ができる。</li>
                            <li>基本的なデータ構造とその操作アルゴリズムについて、C言語を用いてプログラムできる。特に、メモリ管理を必要とするデータ構造の実装を行える。</li>
                        </ul>
                    </td>
                </tr>
                <tr>
                    <th>授業方法</th>
                    <td colspan="7">
                        <div></div>
                        【面接授業】全ての回を対面にて実施します。座席指定はしない。講義は教科書と教科書を補完するパワーポイントを用いて進め、配布資料を追加することもある。パワーポイントと配布資料はPDFをMaNaBoで公開する。講義のほかに、随時、実習を行う。ほぼ毎回の授業で、リアルタイムにMaNaBo上で演習を行い、活動の様子（クイズ集計結果やフォーラム討論の様子）を共有する。各自のコンピュータを使用する実習を行う場合もある。講義後提出〆切の課題については、次の週に解説を行う。
                    </td>
                </tr>
                <tr>
                    <th>活用される授業方法</th>
                    <td colspan="7">
                        <div>
                            <input class="usemethod" type="checkbox" checked /> 板書<br>
                            <input class="usemethod" type="checkbox" checked /> スライド（パワーポイント等）の使用<br>
                            <input class="usemethod" type="checkbox" checked /> 板書・スライド以外の視聴覚教材の使用<br>
                            <input class="usemethod" type="checkbox" /> 個人発表<br>
                            <input class="usemethod" type="checkbox" /> グループ発表<br>
                            <input class="usemethod" type="checkbox" /> ディスカッション・ディベート<br>
                            <input class="usemethod" type="checkbox" /> 実技・実習・実験<br>
                            <input class="usemethod" type="checkbox" /> 学内の教室外施設の利用<br>
                            <input class="usemethod" type="checkbox" /> 校外実習・フィールドワーク<br>
                            <input class="usemethod" type="checkbox" /> その他（内容は「履修者へのコメント」欄を確認してください）<br>
                            <input class="usemethod" type="checkbox" /> 特になし<br>
                        </div>
                    </td>
                </tr>
                <tr>
                    <th>成績評価方法・基準</th>
                    <td colspan="7">
                        <table class="table-bordered">
                            <tr>
                                <td>課題・レポート</td>
                                <td>70%</td>
                            </tr>
                            <tr>
                                <td>定期試験</td>
                                <td>30%</td>
                            </tr>
                        </table>
                    </td>
                </tr>
                <tr>
                    <th>教科書・教材・参考文献</th>
                    <td colspan="7">
                        <table class="table-bordered">
                            <tr>
                                <td class="size8">教科書・教材</td>
                                <td>広瀬貞樹 &nbsp;"あるごりずむ"
                                    近代科学社。MaNaBoのほか、授業用のウェブページを使用する。授業中に使用する電子資料の配布元や課題の提出先のURIは授業中に指定する。</td>
                            </tr>
                            <tr>
                                <td>参考文献</td>
                                <td>
                                    <li class="request_reference_url" data-code="4797304952">Cプログラマのためのアルゴリズムとデータ構造</li>
                                    <li class="request_reference_url" data-code="9784901683999">アルゴリズムとデータ構造</li>
                                    <li class="request_reference_url" data-code="784764905702">
                                        Pythonで体験してわかるアルゴリズムとデータ構造</li>
                                </td>
                            </tr>
                        </table>
                    </td>
                </tr>
                <tr>
                    <th>質問への対応（オフィスアワー等）</th>
                    <td colspan="7">授業後のほか、電子メール（アドレスは授業中に通知）に対応する</td>
                </tr>
                <tr>
                    <th>授業計画</th>
                    <td colspan="7">
                        <table class="table table-bordered">
                            <thead>
                                <tr>
                                    <th class="size3">No.</th>
                                    <th>項目</th>
                                    <th>内容</th>
                                </tr>
                            </thead>
                            <tbody>
                                <tr>
                                    <td>1</td>
                                    <td>マージと分割統治法</td>
                                    <td>マージとマージソートのアルゴリズムを学び、その実装を行う</td>
                                </tr>
                                <tr>
                                    <td>2</td>
                                    <td>クイックソート</td>
                                    <td>クイックソートのアルゴリズムを学び、その実装を行う</td>
                                </tr>
                                <tr>
                                    <td>3</td>
                                    <td>構造体、ポインターによるデータ構造の定義</td>
                                    <td>構造体やポインタを用いるデータ構造の定義と例題プログラムを学ぶ</td>
                                </tr>
                                <tr>
                                    <td>4</td>
                                    <td>リスト構造</td>
                                    <td>リスト構造とは何か、どのような情報を表現するのに適しているのかを学ぶ</td>
                                </tr>
                                <tr>
                                    <td>5</td>
                                    <td>リスト構造の実現</td>
                                    <td>リスト構造に関する例題プログラムを作成する</td>
                                </tr>
                                <tr>
                                    <td>6</td>
                                    <td>スタックとキュー</td>
                                    <td>スタックとキューとは何か、どのような情報を表現するのに適しているのかを学ぶ</td>
                                </tr>
                                <tr>
                                    <td>7</td>
                                    <td>スタックとキューの実現</td>
                                    <td>スタックとキューを用いた例題プログラムを作成する</td>
                                </tr>
                                <tr>
                                    <td>8</td>
                                    <td>グラフ</td>
                                    <td>グラフとは何か、どのような情報を表現するのに適しているのかを学ぶ</td>
                                </tr>
                                <tr>
                                    <td>9</td>
                                    <td>グラフの基本探索アルゴリズム</td>
                                    <td>グラフ上の基本探索アルゴリズムを学ぶ</td>
                                </tr>
                                <tr>
                                    <td>10</td>
                                    <td>グラフの基本探索アルゴリズムの実現</td>
                                    <td>基本探索アルゴリズムの例題プログラムを作成する</td>
                                </tr>
                                <tr>
                                    <td>11</td>
                                    <td>洗練された探索アルゴリズム</td>
                                    <td>重み付きグラフや評価関数を利用する探索法を学ぶ</td>
                                </tr>
                                <tr>
                                    <td>12</td>
                                    <td>木構造と二分探索木</td>
                                    <td>木構造と二分探索木とは何か、その特徴とどのような情報を表現するのに適しているのかを学ぶ</td>
                                </tr>
                                <tr>
                                    <td>13</td>
                                    <td>二分探索木の実現と基本操作</td>
                                    <td>二分探索木に関する基本操作アルゴリズムを学び、プログラミングを行う</td>
                                </tr>
                                <tr>
                                    <td>14</td>
                                    <td>ヒープ</td>
                                    <td>ヒープとは何か、どのような情報を表現するのに適しているのかを学ぶ</td>
                                </tr>
                                <tr>
                                    <td>15</td>
                                    <td>ヒープの実現と基本操作</td>
                                    <td>ヒープに関する基本操作アルゴリズムを学び、プログラミングを行う </td>
                                </tr>
                            </tbody>
                        </table>
                    </td>
                </tr>
                <tr>
                    <th>履修者へのコメント</th>
                    <td colspan="7">
                        2では、構造体、ポインターとアドレス、メモリ管理、再帰を利用するＣ言語プログラミングが必須です。プログラミング演習には、十分な自習時間を確保すること。MaNaBoを利用して質問の機会を準備します。TAの皆さんにどんどん質問して、アドバイスをもらってください。
                    </td>
                </tr>
                <tr>
                    <th>事前事後学習</th>
                    <td colspan="7">授業の前後に、毎回のテーマについて教科書の該当ページに目を通し、専門的な記述法と形式的な表現法を確認すること。特に授業後は教科書の文章を注意深く読みなおしておくこと
                        (1時間程度)。ほぼ毎週プログラミングを中心とした課題がありレポートの作成を行う。提出までに1時間程度を予定すること。課題返却時に指示やコメントがあった場合は、次の課題に必ず反映させること。<br />
                    </td>
                </tr>
            </table>
            <div class="margin-bottom-md padding-md" style="border: solid 1px #f00;">
                各授業科目は、各学部・学科・研究科の定める学位授与方針（ディプロマ・ポリシー）と教育課程の構成・実施方針（カリキュラム・ポリシー）に基づき、カリキュラム上に配置されています。学位授与方針と各授業科目との関連については、<a
                    href="https://www.chukyo-u.ac.jp/information/activity/e4.html" target="_blank"
                    rel="noopener">カリキュラム・マップ</a>に掲載されています。
            </div>
            <script>
                $(function () {
                    requestReferenceUrl();
                    $('input[type=checkbox].usemethod').off().on('click', function () {
                        return false;
                    });
                });
            </script>
        </div>
    </div>
</div>
