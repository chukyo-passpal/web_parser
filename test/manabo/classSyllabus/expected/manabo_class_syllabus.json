{
    "object": "「アルゴリズムとデータ構造１」の学習内容を深め、様々なデータ構造と、それらを取り扱う基本的なアルゴリズムの理解を通して、アルゴリズムの設計や解析をするための基本的な手法の修得を指導する。この授業の目標は、実際にコンピュータプログラムを作成する際に、問題に適切なデータ構造とアルゴリズムを利用、選択できるようになることである。授業は講義形式およびプログラミング演習を通して行う。アルゴリズムとデータ構造2では、主にリスト、木などの基本的なデータ構造の利用方法・実現方法を解説する。",
    "goal": [
        "基本的なデータ構造とその操作アルゴリズムについて、コンピュータモデルと情報数学に基づく説明と分析ができる。",
        "基本的なデータ構造とその操作アルゴリズムについて、C言語を用いてプログラムできる。特に、メモリ管理を必要とするデータ構造の実装を行える。"
    ],
    "method": "【面接授業】全ての回を対面にて実施します。座席指定はしない。講義は教科書と教科書を補完するパワーポイントを用いて進め、配布資料を追加することもある。パワーポイントと配布資料はPDFをMaNaBoで公開する。講義のほかに、随時、実習を行う。ほぼ毎回の授業で、リアルタイムにMaNaBo上で演習を行い、活動の様子（クイズ集計結果やフォーラム討論の様子）を共有する。各自のコンピュータを使用する実習を行う場合もある。講義後提出〆切の課題については、次の週に解説を行う。",
    "usedMethods": ["板書", "スライド（パワーポイント等）の使用", "板書・スライド以外の視聴覚教材の使用"],
    "evaluation": [
        { "type": "課題・レポート", "weight": "70%" },
        { "type": "定期試験", "weight": "30%" }
    ],
    "textbooks": [
        {
            "type": "教科書・教材",
            "title": "広瀬貞樹 \"あるごりずむ\" 近代科学社。MaNaBoのほか、授業用のウェブページを使用する。授業中に使用する電子資料の配布元や課題の提出先のURIは授業中に指定する。"
        }
    ],
    "references": [
        { "title": "Cプログラマのためのアルゴリズムとデータ構造", "code": "4797304952" },
        { "title": "アルゴリズムとデータ構造", "code": "9784901683999" },
        { "title": "Pythonで体験してわかるアルゴリズムとデータ構造", "code": "784764905702" }
    ],
    "officeHour": "授業後のほか、電子メール（アドレスは授業中に通知）に対応する",
    "plan": [
        { "no": 1, "item": "マージと分割統治法", "content": "マージとマージソートのアルゴリズムを学び、その実装を行う" },
        { "no": 2, "item": "クイックソート", "content": "クイックソートのアルゴリズムを学び、その実装を行う" },
        { "no": 3, "item": "構造体、ポインターによるデータ構造の定義", "content": "構造体やポインタを用いるデータ構造の定義と例題プログラムを学ぶ" },
        { "no": 4, "item": "リスト構造", "content": "リスト構造とは何か、どのような情報を表現するのに適しているのかを学ぶ" },
        { "no": 5, "item": "リスト構造の実現", "content": "リスト構造に関する例題プログラムを作成する" },
        { "no": 6, "item": "スタックとキュー", "content": "スタックとキューとは何か、どのような情報を表現するのに適しているのかを学ぶ" },
        { "no": 7, "item": "スタックとキューの実現", "content": "スタックとキューを用いた例題プログラムを作成する" },
        { "no": 8, "item": "グラフ", "content": "グラフとは何か、どのような情報を表現するのに適しているのかを学ぶ" },
        { "no": 9, "item": "グラフの基本探索アルゴリズム", "content": "グラフ上の基本探索アルゴリズムを学ぶ" },
        { "no": 10, "item": "グラフの基本探索アルゴリズムの実現", "content": "基本探索アルゴリズムの例題プログラムを作成する" },
        { "no": 11, "item": "洗練された探索アルゴリズム", "content": "重み付きグラフや評価関数を利用する探索法を学ぶ" },
        { "no": 12, "item": "木構造と二分探索木", "content": "木構造と二分探索木とは何か、その特徴とどのような情報を表現するのに適しているのかを学ぶ" },
        { "no": 13, "item": "二分探索木の実現と基本操作", "content": "二分探索木に関する基本操作アルゴリズムを学び、プログラミングを行う" },
        { "no": 14, "item": "ヒープ", "content": "ヒープとは何か、どのような情報を表現するのに適しているのかを学ぶ" },
        { "no": 15, "item": "ヒープの実現と基本操作", "content": "ヒープに関する基本操作アルゴリズムを学び、プログラミングを行う" }
    ],
    "comment": "2では、構造体、ポインターとアドレス、メモリ管理、再帰を利用するＣ言語プログラミングが必須です。プログラミング演習には、十分な自習時間を確保すること。MaNaBoを利用して質問の機会を準備します。TAの皆さんにどんどん質問して、アドバイスをもらってください。",
    "prePostStudy": "授業の前後に、毎回のテーマについて教科書の該当ページに目を通し、専門的な記述法と形式的な表現法を確認すること。特に授業後は教科書の文章を注意深く読みなおしておくこと(1時間程度)。ほぼ毎週プログラミングを中心とした課題がありレポートの作成を行う。提出までに1時間程度を予定すること。課題返却時に指示やコメントがあった場合は、次の課題に必ず反映させること。"
}
